demes/load_dump.py

```diff
@@ -2,39 +2,30 @@
 Functions to load and dump graphs in YAML and JSON formats.
 """
 import json
-import copy
+import io
 
 import ruamel.yaml
 
 import demes
 
 
 def _loads_yaml_asdict(string):
-    with ruamel.yaml.YAML(typ="safe", pure=True) as yaml:
-        data = yaml.load(string)
-    # split comma separated fields
-    for deme in data.get("demes", []):
-        if "ancestors" in deme:
-            deme["ancestors"] = [anc.strip() for anc in deme["ancestors"].split(",")]
-        if "proportions" in deme:
-            deme["proportions"] = [float(p) for p in deme["proportions"].split(",")]
-    for mig in data.get("migrations", {}).get("symmetric", []):
-        mig["demes"] = [deme.strip() for deme in mig["demes"].split(",")]
-    return data
+    with ruamel.yaml.YAML(typ="safe") as yaml:
+        return yaml.load(string)
 
 
 def _dumps_yaml_fromdict(data):
-    data = copy.deepcopy(data)
-    # collapse comma separated fields
-    for deme in data.get("demes", []):
-        if "ancestors" in deme:
-            deme["ancestors"] = ", ".join(deme["ancestors"])
-        if "proportions" in deme:
-            deme["proportions"] = ", ".join([str(p) for p in deme["proportions"]])
-    for mig in data.get("migrations", {}).get("symmetric", []):
-        mig["demes"] = ", ".join(mig["demes"])
-    string = ruamel.yaml.dump(data)
-    return string
+    stream = io.StringIO()
+    with ruamel.yaml.YAML(typ="safe", output=stream) as yaml:
+        # Disable JSON-style inline arrays and dicts.
```

Comment:
How undocumented are these APIs, and how likely are they to lead to breakage over time do you think? Might be worth dropping in comment here about which of these you think might cause problems.

Reply:
There's no reference documentation for the API at all. There are a few examples scattered over https://yaml.readthedocs.io/. The [ruamel pypi page](https://pypi.org/project/ruamel.yaml/) says "Currently existing functionality will throw a warning before being changed/removed.", so we should find out about likely problems before users do. Some of these flags are just making args from the pyyaml load()/dump() functions available via the new API, but the pyyaml docs also aren't clear (and pyaml warns that [API stability is not guaranteed!](https://pyyaml.org/wiki/PyYAMLDocumentation)). I've mostly been hunting through the ruamel code to figure things out.